src/data/markdown/translated-guides/en/02 Using k6/06 Test life cycle.md

```diff
@@ -232,3 +241,62 @@ It's best to think that each stage and each VU has access to a fresh "copy" of w
 It would be extremely complicated and computationally intensive to pass mutable data between all VUs and then to teardown, especially in distributed setups.
 This would go against a core k6 goal: the same script should be executable in multiple modes.
 
+## Additional lifecycle functions
+
+k6 has a few additional ways to use lifecycle functions:
+
+- **`handleSummary()`**. If you want to make a custom summary, k6 calls one more lifecycle function at the very end of the test.
+
+  For details, refer to [Custom summary](/results-output/end-of-test/custom-summary/).
+
+- **Scenario functions**. Instead of the `default` function, you can also run VU code in scenario functions.
+
+  <CodeGroup labels={["scenario-function.js"]} lineNumbers={[]} showCopyButton={[true]}>
+
+  ```javascript
+  import http from 'k6/http';
+  import { sleep } from 'k6';
+
+  export const options = {
+    scenarios: {
+      my_web_test: {
+        // the function this scenario will execute
+        exec: 'webtest',
+        executor: 'constant-vus',
+        vus: 50,
+        duration: '1m',
+      },
+    },
+  };
+
+  export function webtest() {
+    http.get('https://test.k6.io/contacts.php');
+    sleep(Math.random() * 2);
+  }
+  ```
+
+  </CodeGroup>
+
+## Design reasons for k6 stages
+
+The architecture of the k6 is designed to be stable, portable, and modular.
+The separation of the runtime into stages and functions serves these key design goals, providing:
+
+* Stabler, more meaningful test results.
+
+  As `init` code is separated, it doesn't generate metrics.
+  This isolation removes irrelevant computation from the actual VU requests, making test results more accurate.
+
+* Portable scripts
+
+  The separation lets k6 create an `archive` and run the script in a different execution mode, e.g. in the cloud.
+  The init context already defines which files to load and which modules to import.
+
+  In more sophisticated execution, like clustered systems, this separation brings further performance benefits&mdash;certain nodes don't even need writable filesystems.
+  Everything can be kept in memory.
+
+* Reusable functions
```

Comment:
This section is about test modularization. It can be removed and will be mentioned on other sections when we expand more about this topic.